lib/xlat_tables_v2/xlat_tables_core.c

```diff
@@ -1146,6 +1146,10 @@ void __init init_xlat_tables_ctx(xlat_ctx_t *ctx)
 
 	mmap_region_t *mm = ctx->mmap;
 
+	assert(PLAT_VIRT_ADDR_SPACE_SIZE >= MIN_VIRT_ADDR_SPACE_SIZE);
```

Comment:
The virtual address space will depend on the context being used. Hence PLAT_VIRT_ADDR_SPACE_SIZE is not right to be checked here.

va_max_address in the ctx needs to be checked. Also MIN_VIRT_ADDR_SPACE_SIZE will depend on whether TTST is present or not.